go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -0,0 +1,321 @@
+
+	ef := newEgressFirewall(egressFirewall)
+	nsInfo.egressFirewallPolicy = ef
+	var errList []error
+	//lock the newgressFirewall and unlock nsInfo
+	for i, egressFirewallRule := range egressFirewall.Spec.Rules {
+		//process Rules into egressFirewallRules for egressFirewall struct
+		efr, err := newEgressFirewallRule(egressFirewallRule, i)
+		if err != nil {
+			errList = append(errList, fmt.Errorf("error: cannot create EgressFirewall Rule for destination %s to namespace %s - %v",
+				egressFirewallRule.To.CIDRSelector, egressFirewall.Namespace, err))
+			continue
+
+		}
+		ef.egressRules = append(ef.egressRules, efr)
+	}
+	if len(errList) > 0 {
+		return errList
+	}
+
+	existingNodes, err := oc.kube.GetNodes()
+	if err != nil {
+		return []error{fmt.Errorf("error unable to add egressfirewall %s, cannot list nodes: %s", egressFirewall.Name, err)}
+	}
+	for _, node := range existingNodes.Items {
+		joinSwitch := joinSwitch(node.Name)
+		err = ef.addACLToJoinSwitch(joinSwitch, nsInfo.addressSet.GetIPv4HashName(), nsInfo.addressSet.GetIPv6HashName())
+		if err != nil {
+			errList = append(errList, err)
+		}
+	}
+	if errList != nil {
+		return errList
+	}
+
+	egressFirewall.Status.EgressFirewallObjectStatus = "EgressFirewall Rules applied"
+	err = oc.updateEgressFirewallObjectWithRetry(egressFirewall)
+	if err != nil {
+		return []error{err}
+	}
+	return nil
+
+}
+
+func (oc *Controller) updateEgressFirewall(oldEgressFirewall, newEgressFirewall *egressfirewallapi.EgressFirewall) []error {
+	errList := oc.deleteEgressFirewall(oldEgressFirewall)
+	errList = append(errList, oc.addEgressFirewall(newEgressFirewall)...)
+	return errList
+}
+
+func (oc *Controller) deleteEgressFirewall(egressFirewall *egressfirewallapi.EgressFirewall) []error {
+	klog.Infof("Deleting egress Firewall %s in namespace %s", egressFirewall.Name, egressFirewall.Namespace)
+
+	nsInfo, err := oc.waitForNamespaceLocked(egressFirewall.Namespace)
+	if err != nil {
+		return []error{fmt.Errorf("failed to wait for namespace %s event (%v)",
+			egressFirewall.Namespace, err)}
+	}
+	defer nsInfo.Unlock()
+
+	// copy the egressFirewall object out of nsInfo and clear it so an error does not prevent future egressFirewalls
+	// from admission
+	egressFirewallCopy := nsInfo.egressFirewallPolicy
+	nsInfo.egressFirewallPolicy = nil
+
+	existingNodes, err := oc.kube.GetNodes()
+	if err != nil {
+		return []error{fmt.Errorf("error deleting egressFirewall for namespace %s, cannot get nodes to delete ACLS %v",
+			egressFirewallCopy.namespace, err)}
+	}
+
+	stdout, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading",
+		"--columns=_uuid,match", "find", "ACL",
+		fmt.Sprintf("external-ids:egressFirewall=%s", egressFirewallCopy.namespace))
+	if err != nil {
+		return []error{fmt.Errorf("error executing find ACL command, stderr: %q, %+v", stderr, err)}
+	}
+
+	var errList []error
+	splitStdout := strings.Split(stdout, "\n")
+	for index := 0; index < len(splitStdout); index = index + 2 {
```

Comment:
Could this be re-worked to be more understandable? Or alternatively have an example comment of what the output is which this code will parse? Why `index + 2` for example? Why `strings.Contains(splitStdout[index+1], "ip4")`?

Reply:
I added a comment explaining what the code is doing I don't think I can make it cleaner

Reply:
@JacobTanenbaum I think tim has a point. we're just parsing this to get `addressSetHash` and that's only used for an error message. I think we should just get the UUID, skip the complicated parsing, and delete the ACL.

Reply:
I took it out